File: client/Services/IpService.cpp
#include "IpService.h"
#include "../Helpers/HttpClient.h"
#include "../Helpers/JsonParser.h"
#include <stdexcept>

std::string IpService::GetPublicIp() {
    auto response = HttpClient::Get("http://ip-api.com/json/");

    if (response.statusCode != 200) {
        throw std::runtime_error("Failed to fetch IP address");
    }

    auto json = JsonParser::Parse(response.body);
    auto it = json.find("query");
    if (it == json.end()) {
        throw std::runtime_error("Failed to parse IP address from response");
    }

    return it->second;
}

File: client_cs/Services/IpService.cs
using System;
using System.Net.Http;
using System.Text.Json;
using System.Threading.Tasks;
using client.Models;

namespace client.Services
{
    public class IpService
    {
        private readonly HttpClient _client;
        private const string IP_API_URL = "http://ip-api.com/json/";

        public IpService()
        {
            _client = new HttpClient();
        }

        public async Task<string> GetPublicIpAsync()
        {
            try
            {
                // Console.WriteLine("Fetching IP address...");
                var response = await _client.GetAsync(IP_API_URL);
                response.EnsureSuccessStatusCode();
                
                var content = await response.Content.ReadAsStringAsync();
                // Console.WriteLine($"API Response: {content}");
                
                var ipInfo = JsonSerializer.Deserialize<IpApiResponse>(content);
                // Console.WriteLine($"Parsed IP: {ipInfo.Query}");
                
                
                return ipInfo.Query;
            }
            catch (Exception ex)
            {
                // Console.WriteLine($"Error fetching IP: {ex.Message}");
                throw new Exception($"Failed to fetch IP address: {ex.Message}");
            }
        }
    }
}
